subprojects/core/src/main/java/org/gradle/process/internal/JavaExecHandleBuilder.java

```diff
@@ -311,28 +319,47 @@ protected ArgsEnvironment getEffectiveArgsEnvironment() {
         Map<String, String> environment = getActualEnvironment();
 
         if (hasCommandLineExceedMaxLength(getExecutable(), arguments)) {
-            if (shouldUseEnvironmentVariable(arguments, environment)) {
-                int classPathFlagIndex = arguments.indexOf("-cp");
-                environment.put("CLASSPATH", arguments.get(classPathFlagIndex + 1));
-                arguments.remove(classPathFlagIndex);
-                arguments.remove(classPathFlagIndex);
+            int classPathFlagIndex = arguments.indexOf("-cp");
+            if (shouldTryShorteningCommandLine(arguments)) {
+                if (shouldUseEnvironmentVariable(arguments, environment)) {
```

Comment:
I think we should just skip the environment variable option.  The pathing jar looks better in every other way. It's one less thing we have to test for/worry about and it doesn't interfere with someone who might somehow rely on passing things via `CLASSPATH`.

Reply:
Sure, let me remove this.